src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/RegexNodeToSymbolicConverter.cs

```diff
@@ -421,6 +432,34 @@ SymbolicRegexNode<BDD> ConvertSetloop(RegexNode node, bool isLazy)
                 SymbolicRegexNode<BDD> body = _builder.MkSingleton(moveCond);
                 return _builder.MkLoop(body, isLazy, node.M, node.N);
             }
+
+            // TODO: recognizing strictly only [] (RegexNode.Nothing), for example [0-[0]] would not be regonized
+            bool IsNothing(RegexNode node) => node.Type == RegexNode.Nothing || (node.Type == RegexNode.Set && ConvertSet(node).IsNothing);
+
+            bool IsDotStar(RegexNode node) => node.Type == RegexNode.Setloop && Convert(node, false).IsDotStar;
+
+            bool IsIntersect(RegexNode node) => node.Type == RegexNode.Testgroup && IsNothing(node.Child(2));
+
+            bool TryGetIntersection(RegexNode node, List<RegexNode> conjuncts)
```

Comment:
Rather than having the caller always allocate the `List<RegexNode>`, even though it'll only be used when TryGetIntersection returns true, it seems like it'd be better to make the parameter be `[NotNullWhen(true)] out List<RegexNode>? conjuncts`, and have the method set it to null when returning false and `new()` when returning true and populating it.

Reply:
will do